ethereumj-core/src/main/java/org/ethereum/solidity/compiler/SolidityCompiler.java

```diff
@@ -38,38 +47,96 @@ public SolidityCompiler(SystemProperties config) {
         solc = new Solc(config);
     }
 
-    public static Result compile(File sourceDirectory, boolean combinedJson, Options... options) throws IOException {
+    public static Result compile(File sourceDirectory, boolean combinedJson, Option... options) throws IOException {
         return getInstance().compileSrc(sourceDirectory, false, combinedJson, options);
     }
 
-    public enum Options {
+    public static final class Options {
+        public static final OutputOption AST = OutputOption.AST;
+        public static final OutputOption BIN = OutputOption.BIN;
+        public static final OutputOption INTERFACE = OutputOption.INTERFACE;
+        public static final OutputOption ABI = OutputOption.ABI;
+        public static final OutputOption METADATA = OutputOption.METADATA;
+        public static final OutputOption ASTJSON = OutputOption.ASTJSON;
+    }
+
+    public interface Option extends Serializable {
+        String getValue();
+        String getName();
+    }
+
+    public static class ListOption implements Option {
```

Comment:
This code currently designed only for `--allow-paths`, maybe make it just AllowPathsOption or at least make `AllowPathsOption extends ListOption` or any other shortcut, so user will not need to write `ListOption("allow-paths", `

Reply:
You are absolutely right; I'll do the necessary changes.